src/vs/workbench/services/editor/browser/editorService.ts

```diff
@@ -579,14 +600,24 @@ export class EditorService extends Disposable implements EditorServiceImpl {
 
 			// Override handling: pick editor or open specific
 			if (resolvedOptions?.override === EditorOverride.PICK || typeof resolvedOptions?.override === 'string') {
-				return this.openEditorWith(resolvedOptions.override, resolvedEditor, resolvedOptions, resolvedGroup);
+				const resolvedInputWithOptions = await this.editorOverrideService.resolveEditorOverride(editor as IEditorInput, resolvedOptions, resolvedGroup);
+				if (!resolvedInputWithOptions) {
+					return;
+				}
+				return resolvedGroup.openEditor(resolvedInputWithOptions.editor, resolvedInputWithOptions.options ?? resolvedOptions);
 			}
 
 			// Override handling: ask providers to override
 			if (resolvedOptions?.override !== EditorOverride.DISABLED) {
-				const override = this.doOverrideOpenEditor(resolvedEditor, resolvedOptions, resolvedGroup);
-				if (override) {
-					return override;
+				// This will get cleaned up soon, but since the override service no longer uses the override flow we must check that
+				const resolvedInputWithOptions = await this.editorOverrideService.resolveEditorOverride(editor as IEditorInput, resolvedOptions, resolvedGroup);
```

Comment:
Not a fan of explicit casts, why is this needed?

Reply:
I cast this because my function expects an `IEditorInput`, but there's a chance that it's an `IResourceEditorType`. In that case I cast it which might not be the best solution and I should instead try to clean up the typings? I only really care about the resource, or in the case of the `DiffEditorInput` both resources.

Reply:
Yeah suggest to then write code to correctly distill the resource from either of the types as needed.